src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,62 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_ELEMENTS_LENGTH = 10;
+    private static final int NOT_FOUND_ELEM_IDX = -1;
+
+    private final K[] keys;
+    private final V[] values;
+    private int size = 0;
+
+    @SuppressWarnings("unchecked")
+    public StorageImpl() {
+        this.values = (V[]) new Object[MAX_ELEMENTS_LENGTH];
+        this.keys = (K[]) new Object[MAX_ELEMENTS_LENGTH];
+    }
+
     @Override
     public void put(K key, V value) {
+        int existingKeyIndex = this.getKeyIndex(key);
+
+        if (existingKeyIndex != NOT_FOUND_ELEM_IDX) {
+            if (this.values[existingKeyIndex] == null) {
+                this.size += 1;
```

Comment:
```suggestion
                this.size++;
```
Also you can move it after if-else block